extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts

```diff
@@ -905,6 +911,42 @@ export class PublishDatabaseDialog {
 		return loadProfileButton;
 	}
 
+	private createSaveProfileAsButton(view: azdataType.ModelView): azdataType.ButtonComponent {
+		let saveProfileAsButton: azdataType.ButtonComponent = view.modelBuilder.button().withProps({
+			label: constants.saveProfileAsButtonText,
+			title: constants.saveProfileAsButtonText,
+			ariaLabel: constants.saveProfileAsButtonText,
+			width: cssStyles.PublishingOptionsButtonWidth,
+			height: '25px',
+			secondary: true
+		}).component();
+
+		saveProfileAsButton.onDidClick(async () => {
+			const filePath = await vscode.window.showSaveDialog(
+				{
+					defaultUri: vscode.Uri.file(path.join(this.project.projectFolderPath, `${this.project.projectFileName}_1`)),
+					saveLabel: constants.save,
+					filters: {
+						'Publish Settings Files': ['publish.xml'],
+					}
+				}
+			);
+
+			if (!filePath) {
+				return;
+			}
+
+			if (this.savePublishProfile) {
+				const targetConnectionString = this.targetConnectionTextBox?.value ?? '';
```

Comment:
does `targetDatabaseName` also need similar handling, in case the database name hasn't been chosen?

Reply:
Though it was been set at another place to null, but I am handling it here as well, just for sanity purposes.